src/pages/task/actionButtons/previousCommits/index.tsx

```diff
@@ -51,204 +37,146 @@ export const PreviousCommits: React.FC<PreviousCommitsProps> = ({ taskId }) => {
-          skipOrderNumber,
-          buildVariantOptions: {
-            ...bvOptionsBase,
-            statuses: finishedTaskStatuses,
-          },
-        },
-      });
-    }
-  }, [shouldFetchLastExecuted]); // eslint-disable-line react-hooks/exhaustive-deps
-
-  const Link = useLGButtonRouterLink(link);
-
-  return (
-    <PreviousCommitsWrapper>
-      <StyledSelect
-        size="small"
-        data-cy="previous-commits-select"
-        label="Previous commits for this task"
-        allowDeselect={false}
-        onChange={(v: CommitType) =>
-          dispatch({ type: "setSelectState", selectState: v })
+    >
+      No previous versions available.
+    </Tooltip>
+  ) : (
+    <Menu
+      trigger={
+        <Button rightGlyph={<Icon glyph="CaretDown" />} size={Size.Small}>
+          Previous commits
+        </Button>
+      }
+    >
+      <MenuItem
+        as={Link}
+        disabled={parentLoading}
+        onClick={() =>
+          sendEvent({
+            name: "Submit Previous Commit Selector",
+            type: CommitType.Base,
+          })
         }
-        value={selectState}
-        disabled={!versionMetadata?.baseVersion}
+        to={linkObject[CommitType.Base]}
       >
-        <Option value={CommitType.Base}>
-          Go to {versionMetadata?.isPatch ? "base" : "previous"} commit
-        </Option>
-        <Option value={CommitType.LastPassing}>
-          Go to last passing version
-        </Option>
-        <Option value={CommitType.LastExecuted}>
-          Go to last executed version
-        </Option>
-      </StyledSelect>
-
-      <ConditionalWrapper
-        condition={disableButton}
-        wrapper={(children) => (
-          <Tooltip
-            align="top"
-            justify="middle"
-            triggerEvent="hover"
-            trigger={children}
-          >
-            {loading
-              ? `Fetching...`
-              : `There is no version that satisfies this criteria.`}
-          </Tooltip>
-        )}
+        Go to {versionMetadata?.isPatch ? "base" : "previous"} commit
+      </MenuItem>
+      <MenuItem
+        as={Link}
+        disabled={passingLoading}
+        onClick={() =>
+          sendEvent({
+            name: "Submit Previous Commit Selector",
+            type: CommitType.LastPassing,
+          })
+        }
+        to={linkObject[CommitType.LastPassing]}
       >
-        <LoadingButton
-          as={Link}
-          data-cy="previous-commits-go-button"
-          disabled={disableButton}
-          loading={loading}
-          onClick={() =>
-            sendEvent({
-              name: "Submit Previous Commit Selector",
-              type: selectState,
-            })
-          }
-          size="small"
-        >
-          Go
-        </LoadingButton>
-      </ConditionalWrapper>
-    </PreviousCommitsWrapper>
+        Go to last passing version
+      </MenuItem>
+      <MenuItem
+        as={Link}
+        disabled={executedLoading}
```

Comment:
These can also be disabled if its the first execution of the task. Or if there wasn't one found in the last 300 versions for any of the above tasks.